src/esp/scene/test/ReplicaSceneTest.cpp

```diff
@@ -0,0 +1,106 @@
+#include <Corrade/TestSuite/Compare/Numeric.h>
+#include <Corrade/TestSuite/Tester.h>
+
+#include <Corrade/Utility/Directory.h>
+#include <Magnum/EigenIntegration/GeometryIntegration.h>
+#include <Magnum/EigenIntegration/Integration.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Vector3.h>
+
+#include "configure.h"
+#include "esp/scene/ReplicaSemanticScene.h"
+#include "esp/scene/SemanticScene.h"
+#include "esp/sim/Simulator.h"
+
+#include "esp/assets/GenericInstanceMeshData.h"
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+
+namespace {
+
+const std::string replicaRoom0 =
+    Cr::Utility::Directory::join(SCENE_DATASETS,
+                                 "replica_dataset/room_0/habitat");
+
+struct ReplicaSceneTest : Cr::TestSuite::Tester {
+  explicit ReplicaSceneTest();
+
+  void testSemanticSceneOBB();
+
+  void testSemanticSceneLoading();
+};
+
+ReplicaSceneTest::ReplicaSceneTest() {
+  addTests({&ReplicaSceneTest::testSemanticSceneOBB,
+            &ReplicaSceneTest::testSemanticSceneLoading});
+}
+
+void ReplicaSceneTest::testSemanticSceneOBB() {
+  if (!Cr::Utility::Directory::exists(replicaRoom0)) {
+    CORRADE_SKIP("Replica dataset not found at '" + replicaRoom0 +
+                 "'\nSkipping test");
+  }
+
+  esp::scene::SemanticScene scene;
+  CORRADE_VERIFY(esp::scene::SemanticScene::loadReplicaHouse(
+      Cr::Utility::Directory::join(replicaRoom0, "info_semantic.json"), scene));
+
+  esp::assets::GenericInstanceMeshData mesh;
+  CORRADE_VERIFY(mesh.loadPLY(
+      Cr::Utility::Directory::join(replicaRoom0, "mesh_semantic.ply")));
+
+  const auto& vbo = mesh.getVertexBufferObjectCPU();
+  const auto& objectIds = mesh.getObjectIdsBufferObjectCPU();
+  const auto& ibo = mesh.getIndexBufferObjectCPU();
+
+  CORRADE_VERIFY(objectIds.size() == ibo.size());
+  for (const auto& obj : scene.objects()) {
+    if (obj == nullptr)
+      continue;
+
+    const auto& stringId = obj->id();
+    const int id = std::stoi(stringId.substr(stringId.find_last_of("_") + 1));
+
+    for (uint64_t fid = 0; fid < ibo.size(); fid += 6) {
+      if (objectIds[fid] == id) {
+        esp::vec3f quadCenter = esp::vec3f::Zero();
+        // Mesh was converted from quads to tris
+        for (int i = 0; i < 6; ++i) {
+          quadCenter += vbo[ibo[fid + i]] / 6;
+        }
+
+        CORRADE_VERIFY(obj->obb().contains(quadCenter, 5e-2));
```

Comment:
With #505 merged, you can now use [`CORRADE_ITERATION()`](https://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Tester.html#TestSuite-Tester-iteration-annotations) to have an actually usable diagnostic when something fails. For example:

```cpp
  for (const auto& obj : scene.objects()) {
    if (obj == nullptr)
      continue;

    const auto& stringId = obj->id();
    const int id = std::stoi(stringId.substr(stringId.find_last_of("_") + 1));
    CORRADE_ITERATION(stringId);

    for (uint64_t fid = 0; fid < ibo.size(); fid += 6) {
      CORRADE_ITERATION(fid);
      …
```

Reply:
Ooo yay.  I wanting to use that but wasn't in our repo yet.